Model/Cart/Item/Builder.php

```diff
@@ -144,7 +144,13 @@ public function buildItemId(Item $item)
                 return $parentIds[0];
             }
         }
-        return (string)$item->getProduct()->getId();
+
+        $product = $item->getProduct();
+        $productId = $product ? (string)$product->getId() : '';
```

Comment:
Why not just check that the product is instance of `ProductInterface`?

Reply:
Simplified, thanks